modyn/metadata_database/metadata_database_connection.py

```diff
@@ -95,19 +126,45 @@ def add_selector_state_metadata_trigger(self, pipeline_id: int, trigger_id: int)
             pipeline_id, trigger_id, self.session, self.engine, self.hash_partition_modulus
         )
 
-    def add_trained_model(self, pipeline_id: int, trigger_id: int, model_path: str) -> int:
+    def add_trained_model(
+        self,
+        pipeline_id: int,
+        trigger_id: int,
+        model_path: str,
+        metadata_path: Optional[str] = None,
+        parent_model: Optional[int] = None,
```

Comment:
I agree. `metadata_path` is actually no longer optional. The parent model is None, whenever the model is stored in its entirety (according to a full model strategy). Otherwise it is stored as a delta and has a parent.